compiler/cli/cli-common/src/org/jetbrains/kotlin/cli/common/messages/CompilerMessageExtendedLocation.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.cli.common.messages
+
+import java.io.Serializable
+
+interface CompilerMessageExtendedLocation {
+    val path: String
+    val line: Int
+    val column: Int
+    val lineEnd: Int
+    val columnEnd: Int
+    val lineContent: String?
+
+    companion object {
+        @JvmStatic
+        fun create(path: String?): CompilerMessageExtendedLocation? =
+            create(path, -1, -1, null)
+
+        @JvmStatic
+        fun create(path: String?, line: Int, column: Int, lineContent: String?): CompilerMessageExtendedLocation? =
+            if (path == null) null else CompilerMessageExtendedLocationImpl(path, line, column, -1, -1, lineContent)
+
+        @JvmStatic
+        fun create(
+            path: String?,
+            lineStart: Int,
+            columnStart: Int,
+            lineEnd: Int?,
+            columnEnd: Int?,
+            lineContent: String?
+        ): CompilerMessageExtendedLocation? =
+            if (path == null) null else CompilerMessageExtendedLocationImpl(
+                path,
+                lineStart,
+                columnStart,
+                lineEnd ?: -1,
+                columnEnd ?: -1,
+                lineContent
+            )
+    }
+}
+
+private data class CompilerMessageExtendedLocationImpl(
+    override val path: String,
+    override val line: Int,
+    override val column: Int,
+    override val lineEnd: Int,
+    override val columnEnd: Int,
+    override val lineContent: String?,
+) : Serializable, CompilerMessageExtendedLocation {
```

Comment:
The combination of private and serializable seems suspicious to me. I'd maybe make an interface extending Serializable, although at the moment it seems not needed, and adding it later seems shouldn't break things.

Reply:
Removed Serializable